lib/Twig/Extension/Core.php

```diff
@@ -976,7 +976,12 @@ function twig_escape_filter(Twig_Environment $env, $string, $strategy = 'html',
                 // \uHHHH
                 $char = twig_convert_encoding($char, 'UTF-16BE', 'UTF-8');
 
-                return '\\u'.strtoupper(substr('0000'.bin2hex($char), -4));
+                return implode('', array_map(
```

Comment:
Based on http://ww.ephestione.altervista.org/ and local testing on PHP7.1 `sprintf()` was marginally faster. Updated in 198f4d37c5fe04dfde7eb0d4d4c32772616915db